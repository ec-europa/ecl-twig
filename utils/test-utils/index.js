const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
