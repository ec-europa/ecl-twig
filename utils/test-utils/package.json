{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
