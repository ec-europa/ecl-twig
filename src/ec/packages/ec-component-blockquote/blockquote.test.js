import path from 'path';
import data from '@ecl/ec-specs-blockquote/demo/data';
import { renderTwigFile } from '@ecl-twig/test-utils';

describe('EC - Blockquote', () => {
  test('renders correctly', done => {
    expect.assertions(1);

    const template = path.resolve(__dirname, './blockquote.html.twig');

    renderTwigFile(template, data, (err, html) => {
      expect(html).toMatchSnapshot();
      done();
    });
  });
});
