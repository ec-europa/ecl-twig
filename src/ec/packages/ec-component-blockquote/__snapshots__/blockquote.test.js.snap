// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Blockquote renders correctly 1`] = `
"<blockquote class=\\"ecl-blockquote\\">
  <p class=\\"ecl-blockquote__body\\">
    An interconnected grid will help deliver the ultimate goal of the Energy
    Union, to ensure affordable, secure and sustainable energy, and also growth
    across the EU.
  </p>
  <footer class=\\"ecl-blockquote__attribution\\">
    <cite class=\\"ecl-blockquote__author\\">President Juncker</cite>
  </footer>
</blockquote>
"
`;
